### ISSUE TITLE
Oops; flutter has exited unexpectedly: "Bad state: StreamSink is closed".

### ISSUE THREAD

**Issue by umakanthdiwakar** [flutter/flutter#56804]
Flutter 1.17.0 • channel beta • https://github.com/flutter/flutter.git
Framework • revision e6b34c2b5c (7 days ago) • 2020-05-02 11:39:18 -0700
Engine • revision 540786dd51
Tools • Dart 2.8.1

OS: Windows 10

Consistently hangs with a "... long time", after a new app is created and changes are done within sample app, when run with "flutter run -d chrome". Basically delete the contents in man.dart and put in something resembling this, either with or without a Scaffold within MaterialApp.

main.dart code

```
import 'package:flutter/material.dart';
void main() => runApp(RootWidget());
class RootWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Text("hello flutter")
    );
  }
}
```

**Comment by TahaTesser** [comment]
Hi @umakanthdiwakar 
Not able to reproduce this on latest `beta` channel 
I think you `main.dart` not `man.dart` 

can you please provide your `flutter doctor -v` and `flutter run -d Chrome  --verbose` and make changes in the code and type `r` key to reload and post the logs?
```
For a more detailed help message, press "h". To quit, press "q".
```
Thank you

**Comment by umakanthdiwakar** [comment]
Hi @TahaTesser ,

This time, I introduced Scaffold within MaterialApp, hot reloaded, then undid the change to the source as before and hot reloaded by pressing r, it crashed. Attaching a zip file containing the requested info. In the zip file, flutter_03.log is the log associated with this recent run. Others are previous log files. 

Please let me know if any other info is needed.
[flutissue.zip](https://github.com/flutter/flutter/files/4610693/flutissue.zip)


Thank You.

**Comment by TahaTesser** [comment]
Looks like this was fixed in [#943](https://github.com/dart-lang/webdev/issues/943)

from your crash logs
```
Oops; flutter has exited unexpectedly: "Bad state: StreamSink is closed".

```

Please upgrade to latest master  
Check this [comment](https://github.com/dart-lang/webdev/issues/943#issuecomment-611284544)

already filed
https://github.com/flutter/flutter/issues/54268
https://github.com/flutter/flutter/issues/54661
https://github.com/flutter/flutter/issues/54742
https://github.com/flutter/flutter/issues/55091


if you disagree please write in the comments and I will reopen it 
Thank you

**Comment by umakanthdiwakar** [comment]
Thanks @TahaTesser.
